src/createStore.js

```diff
@@ -159,8 +159,12 @@ export default function createStore(reducer, initialState, enhancer) {
    * @returns {void}
    */
   function replaceReducer(nextReducer) {
-    currentReducer = nextReducer
-    dispatch({ type: ActionTypes.INIT })
+    if (typeof nextReducer === 'function') {
```

Comment:
Nitpicking: let's make it an early exit, like in the beginning of `createStore`.

```js
if (...) {
  throw ...
}

...
```